### PR TITLE
RCAL-776: Change image units.

### LINKED ISSUE #1114: Report L2 data in DN/s instead of e/s
_Issue [RCAL-776](https://jira.stsci.edu/browse/RCAL-776) was created on JIRA by [Eddie Schlafly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eschlafly):_

RTB has requested that we report L2 units in DN/s instead of e/s.  This means:
 * we need to update the L2 schema.
 * We need to rescale the fluxes and variances from ramp fitting into DN/s units.
 ** basically, that means undoing this <https://github.com/spacetelescope/romancal/blob/main/romancal/ramp_fitting/ramp_fit_step.py#L132-L140> and doing the inverse (divide by gain instead of multiply) somewhere around here <https://github.com/spacetelescope/romancal/blob/main/romancal/ramp_fitting/ramp_fit_step.py#L229>

I think we also want to be able to support MJy/sr units for L2 files because that's useful for flux calibration in the L3 step, so when updating the units in the L2 schema, I'd propose making them a union of either DN/s or MJy/sr.  [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham) should weigh in though.

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RCAL-776](https://jira.stsci.edu/browse/RCAL-776)

Closes #1114 #1123 

This PR changed image units in code and tests. Added gain reduction.

NOTE: no regression test link, because these changes require update to regression test files.

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [ ] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Review by ddavis-stsci — CHANGES_REQUESTED** [review]
This should wait until the rad and roman_datamodels changes are in. We don't really want the pyproject to point to your local branches. 
The code changes look fine.

**Comment by PaulHuwe** [comment]
> This should wait until the rad and roman_datamodels changes are in. We don't really want the pyproject to point to your local branches. The code changes look fine.

Yeah, they are temporary just for tests to pass. I have updated the Changelog.

File: romancal/ramp_fitting/ramp_fit_step.py [hunk 1]
```diff
@@ -229,6 +236,12 @@ def ols_cas22(self, input_model, readnoise_model, gain_model):
         image_info = (slopes, ramp_dq, var_poisson, var_rnoise, err)
         image_model = create_image_model(input_model, image_info)
 
+        # Rescale by the gain back to DN/s
+        image_model.data /= gain[4:-4, 4:-4]
+        image_model.err /= gain[4:-4, 4:-4]
+        image_model.var_poisson /= gain[4:-4, 4:-4] ** 2
+        image_model.var_rnoise /= gain[4:-4, 4:-4] ** 2
+
```
Comment (schlafly) [inline romancal/ramp_fitting/ramp_fit_step.py:244]: Staring at the schema, there is also var_flat, which we don't actually populate or do anything with.  I guess that's a separate ticket.

**Review by schlafly — CHANGES_REQUESTED** [review]

File: romancal/ramp_fitting/ramp_fit_step.py (hunk 1 above)
Comment (schlafly) [reply romancal/ramp_fitting/ramp_fit_step.py:244]: This is actually fine and I was somehow missing the flat field uncertainty step computation later on.  Disregard this, sorry!

File: romancal/ramp_fitting/ramp_fit_step.py [hunk 2]
```diff
@@ -157,6 +157,13 @@ def ols(self, input_model, readnoise_model, gain_model):
             )
 
         out_model = create_image_model(input_model, image_info)
+
+        # Rescale by the gain back to DN/s
+        out_model.data /= gain_model.data[4:-4, 4:-4].value
+        out_model.err /= gain_model.data[4:-4, 4:-4].value
+        out_model.var_poisson /= gain_model.data[4:-4, 4:-4].value ** 2
+        out_model.var_rnoise /= gain_model.data[4:-4, 4:-4].value ** 2
+
```
[the remark this answers is not in the document]
Comment (schlafly) [reply romancal/ramp_fitting/ramp_fit_step.py:166]: No, the earlier lines do the exact opposite of what you're doing here?  We plan to get rid of the support for the old ramp fitting mode eventually anyway, but I don't see why we should multiply in the gain and then divide it back out without doing any work in the meantime?  Do you see something I'm missing?

[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/ramp_fitting/ramp_fit_step.py]: Removed.

**Review by ddavis-stsci — APPROVED** [review]
LGTM

**Pull request merged**